File: Lib/Telerik/Examples/QuickStart/ChartView/DrillDown/DrillDownDataInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;

namespace Telerik.Examples.WinControls.ChartView.DrillDown
{
    public class DrillDownDataInfo : INotifyPropertyChanged
    {
        double value;
        DateTime date;

        public DrillDownDataInfo(DateTime date, double value)
        {
            this.date = date;
            this.value = value;
        }

        public double Value
        {
            get
            {
                return this.value;
            }
            set
            {
                this.value = value;
                this.OnPropertyChanged("Value");
            }
        }

        public DateTime Date
        {
            get
            {
                return this.date;
            }
            set
            {
                this.date = value;
                this.OnPropertyChanged("Date");
            }
        }

        public event PropertyChangedEventHandler PropertyChanged;

        public void OnPropertyChanged(string propertyName)
        {
            if (PropertyChanged != null)
            {
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
            }
        }
    }
}

File: Lib/Telerik/Examples/QuickStart/ChartView/DrillDown/DrillDownDataInfo.vb
Imports System.Linq
Imports System.Text
Imports System.ComponentModel

Namespace Telerik.Examples.WinControls.ChartView.DrillDown
	Public Class DrillDownDataInfo
		Implements INotifyPropertyChanged
'INSTANT VB NOTE: The variable value was renamed since Visual Basic does not allow class members with the same name:
		Private value_Renamed As Double
'INSTANT VB NOTE: The variable date was renamed since Visual Basic does not allow class members with the same name:
		Private date_Renamed As Date

		Public Sub New(ByVal [date] As Date, ByVal value As Double)
			Me.date_Renamed = [date]
			Me.value_Renamed = value
		End Sub

		Public Property Value() As Double
			Get
				Return Me.value_Renamed
			End Get
			Set(ByVal value As Double)
				Me.value_Renamed = value
				Me.OnPropertyChanged("Value")
			End Set
		End Property

		Public Property [Date]() As Date
			Get
				Return Me.date_Renamed
			End Get
			Set(ByVal value As Date)
				Me.date_Renamed = value
				Me.OnPropertyChanged("Date")
			End Set
		End Property

		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

		Public Sub OnPropertyChanged(ByVal propertyName As String)
			RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
		End Sub
	End Class
End Namespace
